runtime/src/gpu/amd/gpu-amd.c

```diff
@@ -391,15 +391,15 @@ bool chpl_gpu_impl_can_sort(void){
   return chpl_gpu_impl_can_reduce();
 }
 
-void chpl_gpu_impl_host_register(void* var, size_t size) {
-  // To do this we also need to get a device pointer using
-  // `hipHostGetDevicePointer` and pass the launched kernel
-
-  // hipHostRegister(var, size, hipHostRegisterPortable);
+void* chpl_gpu_impl_host_register(void* var, size_t size) {
+  hipHostRegister(var, size, hipHostRegisterPortable);
+  void *dev_var;
+  hipHostGetDevicePointer(&dev_var, var, 0);
+  return dev_var;
 }
 
 void chpl_gpu_impl_host_unregister(void* var) {
-  // hipHostUnregister(var);
+  hipHostUnregister(var);
```

Comment:
These should have `ROCM_CALL` around them. I also see that they are missing in CUDA versions. Could you add them while here?